lib/pennyworth/cli.rb

```diff
@@ -0,0 +1,339 @@
+# Copyright (c) 2013-2015 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+module Pennyworth
+  class Cli
+    extend GLI::App
+
+    program_desc 'A tool for controlling networks of machines for integration testing'
+    program_long_desc <<-LONGDESC
+      Pennyworth is a tool for controlling a network of machines for
+      integration testing. It helps to control virtual and real machines to
+      provide a well-defined test environment for automated tests.
+
+      Use the global `--definitions-dir` option to specify the path to the
+      directory containing Kiwi and Vagrant definitions. The directory needs to
+      contain `kiwi/` and `vagrant/` subdirectories. Default is `~/.pennyworth`.
+
+      Pennyworth writes a log file to `/tmp/pennyworth.log`.
+
+      Use the `help` command to get documentation about the individual commands.
+      Find more documentation at https://github.com/SUSE/pennyworth.
+    LONGDESC
+
+    preserve_argv(true)
+    @version = Pennyworth::VERSION
+    switch :version, :negatable => false, :desc => "Show version"
+    switch [:help, :h], :negatable => false, :desc => "Show help"
+    switch :verbose, :negatable => false, :desc => "Verbose"
+    switch [:silent], :negatable => false, :desc => "Silent mode"
+    flag ["definitions-dir", :d],
+      :desc => "Path to the directory containing machine definitions",
+      :arg_name => "DEFINITIONS_DIR"
+
+    pre do |global_options,command,options,args|
```

Comment:
Unused block argument - `command`. If it's necessary, use `_` or `_command` as an argument name to indicate that it won't be used.<br>Unused block argument - `options`. If it's necessary, use `_` or `_options` as an argument name to indicate that it won't be used.<br>Unused block argument - `args`. If it's necessary, use `_` or `_args` as an argument name to indicate that it won't be used.<br>Space missing after comma.